astropy/io/registry/core.py

```diff
@@ -362,16 +365,14 @@ def write(self, data, *args, format=None, **kwargs):
             .. versionadded:: 4.3
         """
         # Expand a tilde-prefixed path if present in args[0]
-        args = _expand_user_in_args(args)
+        if len(args):
+            args = (_expand_user_in_arg0(args[0]), *args[1:])
 
         if format is None:
             path = None
             fileobj = None
             if len(args):
                 if isinstance(args[0], PATH_TYPES):
-                    # path might be a os.PathLike object
-                    if isinstance(args[0], os.PathLike):
-                        args = (os.fspath(args[0]),) + args[1:]
```

Comment:
So forcing the conversion to str/bytes is no more necessary ? (are we sure some readers/writers do not rely on it ?)

Reply:
That's not it. The conversion is already handled in the helper function: `os.path.expanduser` never returns a `Path` (that's already true without my refactor), I'm just trimming out redundant bits here.

Reply:
nope:
```
In [4]: _expand_user_in_args((pathlib.Path("foo"),))
Out[4]: (PosixPath('foo'),)
```

Reply:
I see, the issue is that the path isn't resolved, so its "parent" directory might be returned as `''`. Let me fix this too then.